### DISCUSSION TITLE
Translation from prompt format to llama.cpp options?

### DISCUSSION THREAD

**Discussion by Azirine** [ggml-org/llama.cpp#6455] [Q&A]
For example, [here](https://huggingface.co/miqudev/miqu-1-70b/discussions/25#65e5d912ae6f4df165a0dd88) it is said miqu benefits from this prompt format:
>[INST] {System}[/INST]</s>\n[INST] {User}[/INST] {Assistant}

How does it translate in terms of what options to use in llama.cpp instruct mode?
-ins -p ? -r ? --in-prefix ? --in-suffix ?

**Comment by Jeximo** [comment]
```
[INST] {System}[/INST]</s>\n[INST] {User}[/INST] {Assistant}
```

Based on the comments in your link, I **guess** stop token `</s>` and newline `\n` is **not** required after every `[/INST]`.  

Here's an example for `main` CLI: `-i -e --in-prefix "[INST] " --in-suffix "[/INST]" -p "[INST] Below is an instruction that describes a task. Write a response that appropriately completes the request.[/INST]</s>\n[INST] Hello.[/INST]Hi! I will respond to your request."`
